### ISSUE TITLE
Japanese input does not work well right after the URL component

### ISSUE THREAD

**Issue by natsuokawai** [draft-js-plugins/draft-js-plugins#2795]
## Environment
plugin name: draft-js-linkify-plugin
reproducible environment: https://codesandbox.io/s/nz8fj

## Description
The behavior is strange when Japanese is entered right after the URL.

Normally, when you are typing Japanese, the string you are typing is placed in the `<span>` tag, even if it editorState is inComositionMode.
![screenshot1](https://user-images.githubusercontent.com/25678257/170667655-0d8294fc-0f1b-4f67-9114-917bbe1e64e0.png)

However, if you type Japanese right after a URL, a string of text that is not enclosed in a tag is inserted after the `<a>` tag, 

![screenshot2](https://user-images.githubusercontent.com/25678257/170667820-7a4df3ea-68ad-4fb9-80d8-713e3b97dcdc.png)

And when you press Return to confirm the conversion, the string is again placed inside the `<span>` tag. **It causes characters to be typed twice!**
![screenshot3](https://user-images.githubusercontent.com/25678257/170668250-2afbf7b6-e496-499a-a5dd-75b5bfd2781c.png)

**Comment by fxOne** [comment]
This plugin uses `linkify-it` to extract the link from the Text as you can see here: https://github.com/draft-js-plugins/draft-js-plugins/blob/896975c5ee7702de67c9c218f83b1c2aba26e014/packages/linkify/src/utils/extractLinks.ts#L15
As the function as also exported from the plugin, can you please test if the wrong result comes from the `linkify-it` as I have no experience with Japanese characters?

**Comment by natsuokawai** [comment]
@fxOne 
Thank you.

I tried it as you suggested and found that `linkify-it` gave the correct result (no double letters). I also found the following

First of all, to confirm, the character input under Japanese input mode takes the following steps. 1.

Step 1. input a key and a character appears on the screen (enter [Composition mode](https://developer.mozilla.org/ja/docs/Web/API/Element/compositionstart_event))
Step 2. convert the string of characters into Kanji characters, etc.
Step3. press return key to confirm the character (Composition mode ends)

In this issue, the text is inserted outside the `<a>` tag at Step1 (See the second image above.), and linkify-it is not processed at this point.
When the return is entered in Step3, the linkify-it process is executed and the return value is the correct result.
This leads us to believe that it is not due to linkify-it returning an incorrect result.

Where is the process that could insert the characters at Step1?

Translated with www.DeepL.com/Translator (free version)
